### PR TITLE
Fix the format of doxygen comments of output arguments

### PR DESCRIPTION
**Pull request by LaszloLango**
JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.c [hunk 1]
```diff
@@ -925,7 +925,7 @@ ecma_date_set_internal_property (ecma_value_t this_arg, /**< this argument */
  * Insert leading zeros to a string of a number if needed.
  */
 void
-ecma_date_insert_leading_zeros (ecma_string_t **str_p, /**< input/output string */
+ecma_date_insert_leading_zeros (ecma_string_t **str_p, /**< [in, out] ecma string */
```
Comment (akiss77) [inline jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.c]: no space between `in, out`

File: jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c [hunk 2]
```diff
@@ -328,7 +328,7 @@ ecma_builtin_helper_array_index_normalize (ecma_number_t index, /**< index */
  */
 ecma_value_t
 ecma_builtin_helper_array_concat_value (ecma_object_t *obj_p, /**< array */
-                                        uint32_t *length_p, /**< in-out: array's length */
+                                        uint32_t *length_p, /**< [in, out] array's length */
```
Comment (akiss77) [inline jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c]: ditto

File: jerry-core/jerry.c [hunk 3]
```diff
@@ -558,7 +558,7 @@ jerry_api_convert_eval_completion_to_retval (jerry_api_value_t *retval_p, /**< o
  */
 ssize_t
 jerry_api_string_to_char_buffer (const jerry_api_string_t *string_p, /**< string descriptor */
-                                 jerry_api_char_t *buffer_p, /**< output characters buffer */
+                                 jerry_api_char_t *buffer_p, /**< [out] characters buffer */
```
Comment (akiss77) [inline jerry-core/jerry.c]: It is arguable whether this "output" was ment to be a direction specifier or a name. The line below refers to this argument as the "output buffer". Alternatively, best of both worlds: `[out] output characters buffer`?

File: jerry-core/jrt/jrt.c [hunk 4]
```diff
@@ -28,9 +28,9 @@
 bool __attr_always_inline___
 jrt_read_from_buffer_by_offset (const uint8_t *buffer_p, /**< buffer */
                                 size_t buffer_size, /**< size of buffer */
-                                size_t *in_out_buffer_offset_p, /**< in: offset to read from,
-                                                                 *   out: offset, incremented on sizeof (T) */
-                                void *out_data_p, /**< out: data */
+                                size_t *in_out_buffer_offset_p, /**< [in, out] offset to read from /
```
Comment (akiss77) [inline jerry-core/jrt/jrt.c]: no space

File: jerry-core/jrt/jrt.c [hunk 5]
```diff
@@ -28,9 +28,9 @@
...
+                                                                  * incremented on sizeof (T) */
```
Comment (akiss77) [inline jerry-core/jrt/jrt.c]: is the leading asterisk placed OK?

File: jerry-core/jrt/jrt.c [hunk 6]
```diff
@@ -56,8 +56,8 @@ jrt_read_from_buffer_by_offset (const uint8_t *buffer_p, /**< buffer */
 bool __attr_always_inline___
 jrt_write_to_buffer_by_offset (uint8_t *buffer_p, /**< buffer */
                                size_t buffer_size, /**< size of buffer */
-                               size_t *in_out_buffer_offset_p, /**< in: offset to read from,
-                                                                *   out: offset, incremented on sizeof (T) */
+                               size_t *in_out_buffer_offset_p,  /**< [in, out] offset to read from /
+                                                                  * incremented on sizeof (T) */
```
Comment (akiss77) [inline jerry-core/jrt/jrt.c]: ditto 2x

File: jerry-core/lit/lit-snapshot.c [hunk 7]
```diff
@@ -34,7 +34,7 @@ static uint32_t
 lit_snapshot_dump (lit_literal_t lit, /**< literal to dump */
                    uint8_t *buffer_p, /**< buffer to dump to */
                    size_t buffer_size, /**< buffer size */
-                   size_t *in_out_buffer_offset_p) /**< in-out: buffer write offset */
+                   size_t *in_out_buffer_offset_p) /**< [in, out] buffer write offset */
```
Comment (akiss77) [inline jerry-core/lit/lit-snapshot.c]: no space

File: jerry-core/lit/lit-snapshot.c [hunk 8]
```diff
@@ -134,14 +134,14 @@ lit_snapshot_dump (lit_literal_t lit, /**< literal to dump */
  *         false - otherwise.
  */
 bool
-lit_dump_literals_for_snapshot (uint8_t *buffer_p, /**< output snapshot buffer */
+lit_dump_literals_for_snapshot (uint8_t *buffer_p, /**< [out] output snapshot buffer */
                                 size_t buffer_size, /**< size of the buffer */
-                                size_t *in_out_buffer_offset_p, /**< in-out: write position in the buffer */
-                                lit_mem_to_snapshot_id_map_entry_t **out_map_p, /**< out: map from literal identifiers
+                                size_t *in_out_buffer_offset_p, /**< [in, out] write position in the buffer */
```
Comment (akiss77) [inline jerry-core/lit/lit-snapshot.c]: ditto

File: jerry-core/lit/lit-strings.c [hunk 9]
```diff
@@ -602,7 +602,7 @@ lit_read_prev_code_unit_from_utf8 (const lit_utf8_byte_t *buf_p, /**< buffer wit
  * @return next code unit
  */
 ecma_char_t
-lit_utf8_read_next (lit_utf8_byte_t **buf_p) /**< in-out:buffer with characters */
+lit_utf8_read_next (lit_utf8_byte_t **buf_p) /**< [in, out]buffer with characters */
```
Comment (akiss77) [inline jerry-core/lit/lit-strings.c]: ditto

File: jerry-core/vm/vm.c [hunk 10]
```diff
@@ -308,7 +308,7 @@ vm_construct_literal_object (vm_frame_ctx_t *frame_ctx_p, /**< frame context */
  *         false - otherwise.
  */
 static bool __attr_always_inline___
-vm_get_implicit_this_value (ecma_value_t *this_value_p) /**< in-out: this value */
+vm_get_implicit_this_value (ecma_value_t *this_value_p) /**< [in, out] this value */
```
Comment (akiss77) [inline jerry-core/vm/vm.c]: final ditto

File: jerry-core/jerry.c (hunk 3 above)
Comment (LaszloLango) [inline jerry-core/jerry.c]: I agree, `[out] output characters buffer` would be better. I'll fix it.

**Comment by LaszloLango** [comment]
@akiss77, fixed.

**Comment by akiss77** [comment]
LGTM (FWIW)

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**